source/http.c

```diff
@@ -92,811 +90,266 @@ void aws_http_load_log_subject_strings(void) {
     }
 }
 
-static char s_upper(char c) {
-    if (c >= 'a' && c <= 'z') {
-        c = (char)(c - ('a' - 'A'));
-    }
-    return c;
-}
-
-/* https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function */
-static uint32_t s_aws_FNV1a(struct aws_byte_cursor cursor) {
-    uint32_t h = (uint32_t)0x811C9DC5;
-    while (cursor.len--) {
-        char c = (char)s_upper(*cursor.ptr++);
-        h = h ^ (uint32_t)c;
-        h = h * (uint32_t)16777619;
-    }
-    return h;
-}
-
-/* Works like memcmp or strcmp, except is case-agonstic. */
-static int s_strcmp_case_insensitive(const char *a, const char *b, size_t key_len) {
-    for (size_t i = 0; i < key_len; ++i) {
-        int d = s_upper(a[i]) - s_upper(b[i]);
-        if (d) {
-            return d;
-        }
-    }
-    return 0;
-}
-
-/*
- * The next four functions were generated from a small program in C, that takes a text-file as
- * input, reads in header keys as strings, and their corresponding enums as strings. The program
- * then spits out a nicely formatted switch statement for all the key-value pairs. The hashed sting
- * is included as a comment after each switch case. Here is the hash program:
+/**
+ * Given array of aws_byte_cursors, init hashtable where...
+ * Key is aws_byte_cursor* (pointing into cursor from array) and comparisons are case-insensitive.
+ * Value is the array index cast to a void*.
  */
-
-#if 0
-#    define _CRT_SECURE_NO_WARNINGS
-#    include <ctype.h>
-#    include <stdint.h>
-#    include <stdio.h>
-#    include <string.h>
-
-uint32_t s_aws_FNV1a(const char* text) {
-    uint32_t h = (uint32_t)0x811C9DC5;
-    while (*text) {
-        char c = *text++;
-        h = h ^ (uint32_t)c;
-        h = h * (uint32_t)16777619;
-    }
-    return h;
-}
-
-char *s_upper(char *s) {
-    char *orig = s;
-    while (*s) {
-        *s = (char)s_upper(*s);
-        ++s;
+static void s_init_case_insensitive_hash_table(
+    struct aws_hash_table *table,
+    struct aws_allocator *alloc,
+    struct aws_byte_cursor *str_array,
+    int start_index,
+    int end_index) {
+
+    int err = aws_hash_table_init(
+        table,
+        alloc,
+        end_index - start_index,
+        aws_hash_byte_cursor_ptr_ignore_case,
+        (aws_hash_callback_eq_fn *)aws_byte_cursor_eq_ignore_case,
+        NULL,
+        NULL);
+    AWS_FATAL_ASSERT(!err);
+
+    for (size_t i = start_index; i < (size_t)end_index; ++i) {
```

Comment:
The idea is to have a lookup from enum->string (the array), and a lookup from string->enum (the hashtable). So I fill up the array where index=enum, then use that to fill the hashtable.

I don't currently **need** to convert in both directions for every type, but I figured I'd set them all up the same in case I needed to go the other direction in the future, and just to be consistent.

Reply:
I see. Maybe just add a comment then :)

Reply:
It's described in the comment atop the function